packages/common/src/projects/HubProject.ts

```diff
@@ -0,0 +1,357 @@
+  }
+  // istanbul ignore next
+  set description(value: string | undefined) {
+    this.entity.description = value;
+  }
+  // istanbul ignore next
+  get boundary(): IHubGeography | undefined {
+    return this.entity.boundary;
+  }
+  // istanbul ignore next
+  set boundary(value: IHubGeography | undefined) {
+    this.entity.boundary = value;
+  }
+  // istanbul ignore next
+  get culture(): string | undefined {
+    return this.entity.culture;
+  }
+  // istanbul ignore next
+  set culture(value: string | undefined) {
+    this.entity.culture = value;
+  }
+  // istanbul ignore next
+  get tags(): string[] {
+    return this.entity.tags;
+  }
+  // istanbul ignore next
+  set tags(value: string[]) {
+    this.entity.tags = value;
+  }
+  // istanbul ignore next
+  get typeKeywords(): string[] | undefined {
+    return this.entity.typeKeywords;
+  }
+  // istanbul ignore next
+  set typeKeywords(value: string[] | undefined) {
+    this.entity.typeKeywords = value;
+  }
+  // istanbul ignore next
+  get url(): string | undefined {
+    return this.entity.url;
+  }
+  // istanbul ignore next
+  set url(value: string | undefined) {
+    this.entity.url = value;
+  }
+  // istanbul ignore next
+  get id(): string {
+    return this.entity.id;
+  }
+
+  // istanbul ignore next
+  get name(): string {
+    return this.entity.name;
+  }
+  // istanbul ignore next
+  set name(value: string) {
+    this.entity.name = value;
+  }
+  // istanbul ignore next
+  get summary(): string | undefined {
+    return this.entity.summary;
+  }
+  // istanbul ignore next
+  set summary(value: string | undefined) {
+    this.entity.summary = value;
+  }
+  // istanbul ignore next
+  get createdDate(): Date {
+    return this.entity.createdDate;
+  }
+  // istanbul ignore next
+  get createdDateSource(): string {
+    return this.entity.createdDateSource;
+  }
+  // istanbul ignore next
+  get updatedDate(): Date {
+    return this.entity.updatedDate;
+  }
+  // istanbul ignore next
+  get updatedDateSource(): string {
+    return this.entity.updatedDateSource;
+  }
+  // istanbul ignore next
+  get type(): string {
+    return this.entity.type;
+  }
+  // istanbul ignore next
+  get source(): string | undefined {
+    return this.entity.source;
+  }
+  // istanbul ignore next
+  set source(value: string | undefined) {
+    this.entity.source = value;
+  }
+  // istanbul ignore next
+  get slug(): string {
+    return this.entity.slug;
+  }
+  // istanbul ignore next
+  set slug(value: string) {
+    this.entity.slug = value;
+  }
+  // istanbul ignore next
+  get orgUrlKey(): string {
+    return this.entity.orgUrlKey;
+  }
+  // istanbul ignore next
+  get layout(): IHubLayout | undefined {
+    return this.entity.layout;
+  }
+  // istanbul ignore next
+  set layout(value: IHubLayout | undefined) {
+    this.entity.layout = value;
+  }
+  // istanbul ignore next
+  get catalogDefinition(): IHubCatalog {
+    return this.entity.catalogDefinition;
+  }
```

Comment:
Ughhhh... having a prop wrapper for all the entity types is onerous. This seems like a major con of using instances instead of managers.

Reply:
agreed that this is an onerous part of this solution; but I don't see this changing a lot, and it make things much easier to work with.

Reply:
This would be horrible for `IHubContent`, like absolute the worst of all worlds. The fns return POJOs, that have to be returned from a fn that composes them in order to emulate computed props, and then that would go into an instance that would then use actual getters that could have been used to compute the props but instead will be just returning the value that was composed.